torch/_inductor/codegen/cpp.py

```diff
@@ -218,44 +218,58 @@ def _print_ModularIndexing(self, expr):
         x = self.paren(self.doprint(x))
         div = self.paren(self.doprint(div))
         mod = self.paren(self.doprint(mod))
-        if div != "1":
-            x = f"({x} / {div})"
+        if div != "1" and div != "1L" and div != "1.0":
+            if expr.is_integer:
+                x = f"div_floor_internal({x}, {div})"
+            else:
+                x = f"div_floor_internal(static_cast<double>({x}), static_cast<double>({div}))"
         return f"static_cast<{INDEX_TYPE}>({x}) % static_cast<{INDEX_TYPE}>({mod})"
 
     def _print_FloorDiv(self, expr):
         x, div = expr.args
         x = self.paren(self.doprint(x))
         div = self.paren(self.doprint(div))
-        return f"std::floor({x} / {div})"
+        if expr.is_integer:
+            return f"div_floor_internal({x}, {div})"
+        return (
+            f"div_floor_internal(static_cast<double>({x}), static_cast<double>({div}))"
+        )
 
     def _print_floor(self, expr):
         assert len(expr.args) == 1
-        return f"std::floor({self._print(expr.args[0])})"
+        r = f"std::floor({self._print(expr.args[0])})"
+        return f"static_cast<{INDEX_TYPE}>({r})" if expr.is_integer else r
```

Comment:
Please replace `INDEX_TYPE` with another name because it does not represent an index value.

Reply:
Change to ```long```.